rasa_addons/core/nlg/bftemplate.py

```diff
@@ -85,11 +89,31 @@ async def generate(
             fallback_language=fallback_language,
         )
         if "language" in message: del message["language"]
+
+        if self.__is_image_available(message, "image"):
+            message["image"] = self.__substitute_url(message, "image", message["image"])
+        if self.__is_image_available(message, "elements"):
+            #loop over the image urls in a carousel
+            for element in message["elements"]:
+                element["image_url"] = self.__substitute_url(element, "image_url", element["image_url"])
+
         metadata = message.pop("metadata", {}) or {}
         for key in metadata: message[key] = metadata[key]
 
         return message
 
+    def __is_image_available(self, message: dict, value: str):
```

Comment:
No double leading underscores in func names. You can use a single leading one if you want.